cmd/cli/app/history/history_list.go

```diff
@@ -0,0 +1,209 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package history
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"google.golang.org/grpc"
+
+	"github.com/stacklok/minder/cmd/cli/app"
+	"github.com/stacklok/minder/cmd/cli/app/common"
+	"github.com/stacklok/minder/internal/db"
+	"github.com/stacklok/minder/internal/util"
+	"github.com/stacklok/minder/internal/util/cli"
+	"github.com/stacklok/minder/internal/util/cli/table"
+	"github.com/stacklok/minder/internal/util/cli/table/layouts"
+	minderv1 "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
+)
+
+var listCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List history",
+	Long:  `The history list subcommand lets you list history within Minder.`,
+	RunE:  cli.GRPCClientWrapRunE(listCommand),
+}
+
+// listCommand is the profile "list" subcommand
+func listCommand(ctx context.Context, cmd *cobra.Command, _ []string, conn *grpc.ClientConn) error {
+	client := minderv1.NewEvalResultsServiceClient(conn)
+	//client := minderv1.NewProfileServiceClient(conn)
+
+	project := viper.GetString("project")
+	profileName := viper.GetString("profileName")
+	entityName := viper.GetString("entityName")
+	entityType := viper.GetString("entityType")
+	evalStatus := viper.GetString("evalStatus")
+	remediationStatus := viper.GetString("remediationStatus")
+	alertStatus := viper.GetString("alertStatus")
+	format := viper.GetString("output")
+
+	// Ensure the output format is supported
+	if !app.IsOutputFormatSupported(format) {
+		return cli.MessageAndError(fmt.Sprintf("Output format %s not supported", format), fmt.Errorf("invalid argument"))
+	}
+
+	// validate the filters which need validation
+	evalStatusFilter, err := valitedFilter(evalStatus, evalStatuses)
+	if err != nil {
+		return err
+	}
+
+	remediationStatusFilter, err := valitedFilter(remediationStatus, remediationStatuses)
+	if err != nil {
+		return err
+	}
+
+	alertStatusFilter, err := valitedFilter(alertStatus, alertStatuses)
+	if err != nil {
+		return err
+	}
+
+	entityTypeFilter, err := valitedFilter(entityType, entityTypes)
+	if err != nil {
+		return err
+	}
+
+	// list all the things
+	resp, err := client.ListEvaluationHistory(ctx, &minderv1.ListEvaluationHistoryRequest{
+		Context:     &minderv1.Context{Project: &project},
+		EntityType:  entityTypeFilter,
+		EntityName:  asFilter(entityName),
+		ProfileName: asFilter(profileName),
+		Status:      evalStatusFilter,
+		Remediation: remediationStatusFilter,
+		Alert:       alertStatusFilter,
+		From:        nil,
+		To:          nil,
+		Cursor:      nil,
```

Comment:
**polish:** I think we should support cursors from the start,it is very easy to go past 25 items in a real scenario.